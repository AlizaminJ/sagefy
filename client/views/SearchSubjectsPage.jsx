import React from 'react'
import { string, shape, arrayOf } from 'prop-types'
import Layout from './components/Layout'
import Icon from './components/Icon'
import Footer from './components/Footer'
import ChooseSubject from './components/ChooseSubject'
import CreateSubject from './components/CreateSubject'

export default function SearchSubjectsPage({
  role,
  query: { q },
  subjects,
  hash,
}) {
  return (
    <Layout
      hash={hash}
      page="SearchSubjectsPage"
      title="Search subjects"
      description="Find the learning subject you are looking for. Anything you want to learn, Sagefy can help you."
    >
      <section className="ta-c">
        <h1>
          What do you want to learn? <Icon i="search" s="h1" />
        </h1>
        <form action="/subjects/search">
          <p>
            <input
              type="search"
              size="40"
              value={q}
              placeholder="example: Music"
              autoFocus={!!q}
              name="q"
            />
          </p>
          <p>
            <button type="submit">
              <Icon i="search" /> Search
            </button>
          </p>
        </form>
      </section>

      {subjects && subjects.length ? (
        <section>
          <h2>
            Choose from one of these subjects <Icon i="subject" s="h2" />
          </h2>
          <ChooseSubject subjects={subjects} level="goal" />
        </section>
      ) : null}

      {q && (
        <section>
          <p>
            <em>
              Not seeing what you want? <Icon i="error" />
            </em>
          </p>
          <details open={!subjects || !subjects.length}>
            <summary>
              <h2 className="d-i">
                You can suggest a new subject <Icon i="subject" s="h2" />
              </h2>
            </summary>

            <CreateSubject role={role} name={q} />
          </details>
        </section>
      )}

      {/* TODO when !q, show popular subjects here */}

      <Footer role={role} />
    </Layout>
  )
}

SearchSubjectsPage.propTypes = {
  hash: string.isRequired,
  role: string,
  query: shape({}).isRequired,
  subjects: arrayOf(
    shape({
      entityId: string.isRequired,
      name: string.isRequired,
      body: string.isRequired,
    })
  ),
}

SearchSubjectsPage.defaultProps = {
  role: 'sg_anonymous',
  subjects: [],
}
