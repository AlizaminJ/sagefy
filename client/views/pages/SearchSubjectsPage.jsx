import React from 'react'
import { string, shape } from 'prop-types'
import Icon from '../components/Icon'
import Footer from '../components/Footer'
import ChooseSubject from '../components/ChooseSubject'
import CreateSubject from '../components/CreateSubject'

export default function SearchSubjectsPage({
  role,
  query: { q },
  searchSubjects,
}) {
  return (
    <div className="SearchSubjectsPage">
      <section className="ta-c">
        <h1>
          What do you want to learn? <Icon i="search" s="xxl" />
        </h1>
        <form action="/search-subjects">
          <p>
            <input
              type="search"
              size="40"
              value={q}
              placeholder="example: Music"
              autoFocus={!!q}
              name="q"
            />
          </p>
          <p>
            <button type="submit">
              <Icon i="search" /> Search
            </button>
          </p>
        </form>
      </section>

      {(searchSubjects && searchSubjects.nodes.length && (
        <section>
          <h2>
            Choose from one of these subjects <Icon i="subject" s="xl" />
          </h2>
          <ChooseSubject subjects={searchSubjects.nodes} />
        </section>
      )) ||
        null}

      {q && (
        <section>
          <p>
            <em>
              Not seeing what you want? <Icon i="error" />
            </em>
          </p>
          <details open={!searchSubjects || !searchSubjects.nodes.length}>
            <summary>
              <h2 className="d-i">
                You can suggest a new subject <Icon i="subject" s="xl" />
              </h2>
            </summary>

            <CreateSubject role={role} name={q} />
          </details>
        </section>
      )}

      {/* TODO when !q, show popular subjects here */}

      <Footer role={role} />
    </div>
  )
}

SearchSubjectsPage.propTypes = {
  role: string,
  query: shape({}).isRequired,
  searchSubjects: shape({}),
}

SearchSubjectsPage.defaultProps = {
  role: 'sg_anonymous',
  searchSubjects: null,
}
