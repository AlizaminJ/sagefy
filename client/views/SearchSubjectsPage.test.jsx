import SearchSubjectsPage from './SearchSubjectsPage'

describe('SearchSubjectsPage', () => {
  it('should render the search subjects page', () => {
    expect(SearchSubjectsPage({ query: {} })).toMatchSnapshot()
  })
})
