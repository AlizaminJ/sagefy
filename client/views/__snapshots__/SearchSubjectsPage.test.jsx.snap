// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSubjectsPage should render the search subjects page 1`] = `
<Layout
  canonical={null}
  description="Find the learning subject you are looking for. Anything you want to learn, Sagefy can help you."
  image={null}
  page="SearchSubjectsPage"
  title="Search subjects"
>
  <section
    className="ta-c"
  >
    <h1>
      What do you want to learn? 
      <Icon
        i="search"
        s="h1"
      />
    </h1>
    <form
      action="/subjects/search"
    >
      <p>
        <input
          autoFocus={false}
          name="q"
          placeholder="example: Music"
          size="40"
          type="search"
        />
      </p>
      <p>
        <button
          type="submit"
        >
          <Icon
            i="search"
            s="m"
          />
           Search
        </button>
      </p>
    </form>
  </section>
  <Footer
    role="sg_anonymous"
  />
</Layout>
`;
